src/builder/build_traits.rs

```diff
@@ -73,9 +73,9 @@ pub trait Container {
     fn add_constant(
         &mut self,
         constant: ops::Const,
-        extensions: ExtensionSet,
+        extensions: impl Into<Option<ExtensionSet>>,
```

Comment:
Slight generalization here @ss2165 if you are happy with that? All existing calls remain legal.

Reply:
I can do something slightly odd in this PR (i.e. explicitly specify an ExtensionSet that is...not what you might expect) if we'd rather keep like this, 'coz either way we should probably at least try to make another PR where we remove this (and use None/open-extensions) everywhere